src/main/java/com/griefcraft/lwc/LWC.java

```diff
@@ -921,49 +922,78 @@ public Player findPlayer(int minX, int maxX, int minY, int maxY, int minZ, int m
     }
 
     /**
-     * Send a locale to a player or console
+     * Check the data of locale
+     * Returns null if invalid
      *
-     * @param sender
-     * @param key
-     * @param args
+     * @param sender CommandSender
+     * @param key key of locale
+     * @param args Character to be rewritten
+     *              Example: %block% --> Chest
+     * @return message or null
      */
-    public void sendLocale(CommandSender sender, String key, Object... args) {
+    private String[] getLocaleMessage(CommandSender sender, String key, Object... args) {
         String[] message; // The message to send to the player
         MessageParser parser = plugin.getMessageParser();
         String parsed = parser.parseMessage(key, args);
 
         if (parsed == null) {
-            return; // Nothing to send
+             return null; // Nothing to send
         }
 
         // message = parsed.split("\\n");
         message = StringUtils.split(parsed, '\n');
 
         // broadcast an event if they are a player
-        if (sender instanceof Player) {
+        // If the key is "prefix", LWCSendLocaleEvent is not called
+        if (sender instanceof Player && !key.equals("prefix")) {
             LWCSendLocaleEvent evt = new LWCSendLocaleEvent((Player) sender, key);
             moduleLoader.dispatchEvent(evt);
 
             // did they cancel it?
             if (evt.isCancelled()) {
-                return;
+                return null;
             }
         }
 
         if (message == null) {
             sender.sendMessage(Colors.Dark_Red + "LWC: " + Colors.White + "Undefined locale: \"" + Colors.Dark_Gray + key
                     + Colors.White + "\"");
-            return;
+            return null;
         }
 
         if (message.length > 0 && message[0].equalsIgnoreCase("null")) {
+            return null;
+        }
+
+        return message;
+    }
+
+    /**
+     * Send a locale to a player or console
+     *
+     * @param sender
+     * @param key
+     * @param args
+     */
+    public void sendLocale(CommandSender sender, String key, Object... args) {
+        // The message to send to the player
+        String[] message = getLocaleMessage(sender, key, args);
+
+        String[] prefix = getLocaleMessage(sender, "prefix");
+
+        if (message == null) {
             return;
         }
 
```

Comment:
Event should be dispatched here, since this is where the message is guaranteed to be sent.